Kinvey.Shared/Store/MultiInsertRequest.cs

```diff
@@ -362,19 +362,17 @@ private async Task<KinveyMultiInsertResponse<T>> HandleMultiInsertRequestAsync(I
             }
             catch (KinveyException exception)
             {
-                if (exception.StatusCode == 500)
+                if (exception.StatusCode == 401)
                 {
-                    response.Entities = new List<T>();
-                    response.Errors = new List<Error>();
-                    for (var index = 0; index < entities.Count(); index++)
-                    {
-                        response.Entities.Add(default(T));
-                        response.Errors.Add(new Error { Code = 0, Errmsg = exception.Message, Index = index });
-                    }
+                    throw;
                 }
-                else
+
+                response.Entities = new List<T>();
+                response.Errors = new List<Error>();
+                for (var index = 0; index < entities.Count(); index++)
                 {
-                    throw;
+                    response.Entities.Add(default(T));
```

Comment:
@vinaygahlawat correct me if I'm wrong, but I believe a general error (meaning not an error returned by one of the items in the array) should be handled as a regular general error (meaning without wrap the error inside an error with `index` property for example) right?

Reply:
@heyzooi It is an interesting question. In the case with a single multi-insert request that fails as a general error this is correct. But what if, in a set of multi-insert requests that are batched, one of the multi-insert operations fail, but the rest succeed. Should we fail the entire operation? Keep in mind that one or several of the batches may have already been processed successfully.

Reply:
oh, sorry I didn't notice that we were talking about a multi batch request. So in this case that's correct, we should wrap the error.